stdlib/public/core/Array.swift

```diff
@@ -1865,3 +1865,163 @@ internal struct _ArrayAnyHashableBox<Element: Hashable>
     return true
   }
 }
+
+/// `Array<Element>` is a countably infinite direct sum (https://en.wikipedia.org/wiki/Direct_sum) 
```

Comment:
This is not a doc comment, but an implementation comment, so it should start with 2 slashes. Also, everything should fit within 80 columns.